---
title: "Introduction to Most Dominant Transcripts"
subtitle: "*Detection of Most Dominant Transcripts*"
author: "TÃ¼lay Karakulak"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction to Most Dominant Transcripts}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Introduction

This guide provides a step-by-step workflow for identifying the Most Dominant Transcripts (MDTs) within a bulk RNA-sequencing dataset. The **MDTToolset** package is designed to work with outputs generated by various RNA-Seq quantification tools, including, but not limited to, Salmon and Kallisto. For the accurate detection of MDTs, users must utilize normalized data values (e.g., TPM values) for bulk RNA-seq data.

# Load MDTToolset
```{r setup, message=FALSE, warning=FALSE}
library(MDTToolset)
```


## Prepare Data
In this section, we utilize the TPM outputs from the Kallisto tool, originating from an (RNA-Seq experiment) [https://www.nature.com/articles/s41597-020-0541-4]. The data is openly accessible and available for download. The MDTToolset includes two conditions as example cases for users. If you already have the quantification data prepared for each condition, you may proceed to the next step. Ensure that the first two column names are 'ENST' and 'ENSG', respectively, followed by the sample names as shown below.

```{r prepareData}
## read TPM data 
cond1 <- MDTToolset::cond1 # molecular retina
cond2 <- MDTToolset::cond2 # peripheral retina 

# visalize the dataframes. Note that you should have knitr R package installed.
knitr::kable(head(cond1,2), caption = 'TPM Matrix')
knitr::kable(head(cond2,2), caption = 'TPM Matrix')

```

## Prepare Files to Remove the Redundant Transcripts
This step is optional but highly recommended.

A challenge in transcript quantification is the presence of redundant transcripts, which are transcripts with identical sequences that yield the same TPM values. This redundancy can obscure the identification of Most Dominant Transcripts (MDTs) by preventing the detection of a singular, dominant transcript for a given gene. To address this issue, our initial step involves eliminating redundant transcripts from the dataset. This process includes assigning TPM values to a single representative transcript per redundant group, based on the comparison of their protein sequences.

Priority is given to transcripts classified as MANE (Matched Annotation from NCBI and EMBL-EBI) in the Ensembl database, designating them as the primary transcript. In cases where no MANE transcript is identified, a transcript is randomly selected to represent the group. This step is particularly crucial when working with tools like Kallisto, which may assign identical TPM values to redundant transcripts. Following this, the calculation of MDTs proceeds using the reassigned TPM values.

To be able to detect redundat transcripts, we need sequences of transcripts. Fasta sequence of transcripts (cdna sequences) can be downloaded from the designated databases (e.g Ensembl)[https://www.ensembl.org/info/data/ftp/index.html]. We provide Ensembl version 104 as an example. In addition, we provide the Ensembl MANE Select Data (v104) to help remove redundant transcripts found in Kallisto output files.  To download your version of the MANE select, navigate to the Ensembl website for the version you used, proceed to the BioMart website, select the fields *Gene Stable ID*, *Transcript Stable ID*, and *RefSeq Match Transcript (MANE Select)*, and download the TSV file. 
Please note that **The user should employ the same version that was used to quantify their transcript abundances to ensure consistency.**
the function `prepare_Seq()` takes fasta and MANE file as input and gives a dataframe showing ENSG (Gene), ENST (Transcript), and ENSP (Protein), and MANE ID, and ENST Sequence.

```{r loadENSG_ENST_data}
# read MANE Select
ensg_enst_ensp_mane_v104 <- MDTToolset::ensg_enst_ensp_mane_v104
# prepare sequences and Mane select transcripts dataframe
ensg_enst_seq <- prepare_Seq(MDTToolset::Homo_sapiens.GRCh38.cdna.all, ensg_enst_ensp_mane_v104)

## Select transcripts having MANE ids
enst_mane_iso <- ensg_enst_ensp_mane_v104[!(is.na(ensg_enst_ensp_mane_v104$RefSeq.match.transcript..MANE.Select.) | ensg_enst_ensp_mane_v104$RefSeq.match.transcript..MANE.Select. == ''),c(1,2,3,4)]
```
 
We merge the transcript TPM expression matrix file with the ensg_enst_seq file that we have prepared in the previous step so that the second column includes the ENSG ids.
```{r mergeDataSets}
cond1_ENST_ENSG <- merge(ensg_enst_seq[,1:2], cond1, by='ENST')
cond2_ENST_ENSG <- merge(ensg_enst_seq[,1:2], cond2, by='ENST')

# visalize the dataframes. Note that you should have knitr R package installed.
knitr::kable(head(cond1_ENST_ENSG,2), caption = 'TPM Matrix')
knitr::kable(head(cond2_ENST_ENSG,2), caption = 'TPM Matrix')
```

As transcripts with the same sequences might have different peptide lengths, we aim to select the ones with the longest sequence when there is a redundancy. This information cannot be accessed directly via Ensembl Biomart, this we need to download the transcripts' peptide sequences and then calculate the length of the peptides. We provide Ensembl version 104 as an example. In case you want to prepare a different version, download ENSG, ENST, peptide sequences from Biomart and use the following function from MDTToolset:

df_biomart_seq_length <- MDTToolset::MDTToolset::prepare_seq_length('mart_export.txt') 
The file 'mart_export.txt' is in the fasta file format:
>ENSG0000000XXXX|ENST0000000KKK
MQRSPLEKASVVSKLFFSWTRPILRKGYRQRLELSDIYQIPSVDSADNLSEKLEREWDRE
LASKKNPKLINALRRCFFWRFMFYGIFLYLGEVTKAVQPLLLGRIIASYDPDNKEERSIA
IYLG*
>ENSG0000000YYYY|ENST0000000ZZZ
MTAEEMKATESGAQSAPLPMEGVDISPKQDEGVLKVIKREGTGTEMPMIGDRVFVHYTGW
LLDGTKFDSSLD

```{r PeptideLength}
# Example dataframe for df_biomart_seq_length 
df_biomart_seq_length <- MDTToolset::df_biomart_seq_selection

knitr::kable(head(df_biomart_seq_length,2), caption = "Transcripts' Peptide Seq Length")
```


## Remove Redundant Transcripts
`remove_redundant()` function takes three inputs, transcript TPM file (having ENST and ENSG columns), ENST sequence file, and Mane select file prepared in the previous step. If there is more than one transcript of a gene with the same sequences, one of them is removed, its TPM value is added to the selected transcript. Additionally, if there is a transcript assigned as `MANE`, this transcript is prioritized over the other ones. If there is no MANE selects for the redundant transcripts, the transcripts with the longest peptide sequence is chosen. The output will be the same as previous TPM file without redundant ones.

```{r removeRedundant}
## Example Usage
#cond1_ENST_ENSG_nonredundant <- remove_redundants(data_matrix = cond1_ENST_ENSG, enst_ensp_sequences = ensg_enst_seq, mane_select =  NULL, peptide_lengths = NULL)
cond1_ENST_ENSG_nonredundant <- remove_redundants(cond1_ENST_ENSG, ensg_enst_seq, enst_mane_iso, df_biomart_seq_length)
cond2_ENST_ENSG_nonredundant <- remove_redundants(cond2_ENST_ENSG, ensg_enst_seq, enst_mane_iso, df_biomart_seq_length)
```


## Find MDTs
The next step involves identifying the Most Dominant Transcripts (MDTs) in the dataset. The find_mdts() function calculates MDTs based on two criteria: a user-defined cutoff and min_exp. The cutoff parameter specifies the enrichment rate between the expressions of the first and second-ranked transcripts. min_exp determines the minimum expression level that transcripts must meet to be considered as MDTs.
```{r calculateMDTs, message=FALSE, warning=FALSE}
cond1_mdts <- find_mdts(cond1_ENST_ENSG_nonredundant, cutoff = 2, min_exp = 0.2)
cond2_mdts <- find_mdts(cond2_ENST_ENSG_nonredundant, cutoff = 2, min_exp = 0.2)
```

## Visualize count of MDTs
We provide a basic boxplot plot using ggplot to plot the number of MDTs in each condition and also bar graph to visualize number of MDTs in each sample. 
### visualize the number of MDTs in each condition

```{r plotMDTcounts, message=FALSE, warning=FALSE}
# Distribution of number of MDTs in each sample
MDT_boxplot(cond1_mdts, cond2_mdts, title = "Distribution of Number of MDTs", x_axis_label = "Condition", y_axis_label = "Number of MDT")
```

### visualize the number of MDTs in each sample
```{r plotMDTdist, message=FALSE, warning=FALSE}
# Distribution of number of MDTs in each sample
```

