<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MDTToolset">
<title>MDT Switching Event and Its Functional Impact • MDTToolset</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="MDT Switching Event and Its Functional Impact">
<meta property="og:description" content="MDTToolset">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MDTToolset</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/MDTToolset_switching-vignette.html">MDT Switching Event and Its Functional Impact</a>
    <a class="dropdown-item" href="../articles/MDTToolset-vignette.html">Introduction to Most Dominant Transcripts</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/KarakulakTulay/MDTToolset/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>MDT Switching Event and Its Functional Impact</h1>
            <h3 data-toc-skip class="subtitle"><em>Detection of Most
Dominant Transcript Switches and Their Functional Impact</em></h3>
                        <h4 data-toc-skip class="author">Tülay
Karakulak</h4>
            
            <h4 data-toc-skip class="date">2024-05-09</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/KarakulakTulay/MDTToolset/blob/HEAD/vignettes/MDTToolset_switching-vignette.Rmd" class="external-link"><code>vignettes/MDTToolset_switching-vignette.Rmd</code></a></small>
      <div class="d-none name"><code>MDTToolset_switching-vignette.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="load-mdttoolset">Load MDTToolset<a class="anchor" aria-label="anchor" href="#load-mdttoolset"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KarakulakTulay/MDTToolset" class="external-link">MDTToolset</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="prepare-input-file">Prepare Input File<a class="anchor" aria-label="anchor" href="#prepare-input-file"></a>
</h3>
<p>The following files are given together with the MDTToolset. If you
would like to see how they are prepared, please see tutorial:
Introduction to Most Dominant Transcripts.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cond1_mdts</span> <span class="op">&lt;-</span> <span class="fu">MDTToolset</span><span class="fu">::</span><span class="va">cond1_mdts</span></span>
<span><span class="va">cond2_mdts</span> <span class="op">&lt;-</span> <span class="fu">MDTToolset</span><span class="fu">::</span><span class="va">cond2_mdts</span></span>
<span><span class="va">cond1_ENST_ENSG_nonredundant</span> <span class="op">&lt;-</span> <span class="fu">MDTToolset</span><span class="fu">::</span><span class="va">cond1_ENST_ENSG_nonredundant</span></span>
<span><span class="va">cond2_ENST_ENSG_nonredundant</span> <span class="op">&lt;-</span> <span class="fu">MDTToolset</span><span class="fu">::</span><span class="va">cond2_ENST_ENSG_nonredundant</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># read MANE Select</span></span>
<span><span class="va">ensg_enst_ensp_mane_v104</span> <span class="op">&lt;-</span> <span class="fu">MDTToolset</span><span class="fu">::</span><span class="va">ensg_enst_ensp_mane_v104</span></span>
<span><span class="co"># prepare sequences and Mane select transcripts dataframe</span></span>
<span><span class="va">ensg_enst_seq</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepare_Seq.html">prepare_Seq</a></span><span class="op">(</span><span class="fu">MDTToolset</span><span class="fu">::</span><span class="va">Homo_sapiens.GRCh38.cdna.all</span>, <span class="va">ensg_enst_ensp_mane_v104</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Select transcripts having MANE ids</span></span>
<span><span class="va">enst_mane_iso</span> <span class="op">&lt;-</span> <span class="va">ensg_enst_ensp_mane_v104</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">ensg_enst_ensp_mane_v104</span><span class="op">$</span><span class="va">RefSeq.match.transcript..MANE.Select.</span><span class="op">)</span> <span class="op">|</span> <span class="va">ensg_enst_ensp_mane_v104</span><span class="op">$</span><span class="va">RefSeq.match.transcript..MANE.Select.</span> <span class="op">==</span> <span class="st">''</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">4</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Select the only ENSTs having ENSP Protein IDs.</span></span>
<span><span class="va">ensp_only</span> <span class="op">&lt;-</span> <span class="va">ensg_enst_ensp_mane_v104</span><span class="op">[</span><span class="op">!</span><span class="va">ensg_enst_ensp_mane_v104</span><span class="op">$</span><span class="va">Protein.stable.ID</span> <span class="op">==</span> <span class="st">""</span>,<span class="op">]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="calculate-mdt-switching-events-in-the-data">Calculate MDT Switching Events in the Data<a class="anchor" aria-label="anchor" href="#calculate-mdt-switching-events-in-the-data"></a>
</h3>
<p>The <code><a href="../reference/switch_calculator.html">switch_calculator()</a></code> function identifies MDT
switching events in each sample by comparing it to a “Normal” cohort or
another specified cohort, thereby defining dMDT (disease-specific MDTs).
It considers the following criteria:</p>
<ul>
<li>cutoff: The percentage of samples in the comparison group (cond2 in
our case) that may have the same MDT as the Normal cohort. A cutoff of 0
is very stringent and ensures that the dMDT is not observed in any of
the samples from cond2.</li>
<li>cutoff_rate: The enrichment rate between Transcript 1 and Transcript
2 in the cond1_mdts and cond2_mdts datasets.</li>
<li>cutoff_other_MDTs: The percentage of cond2 samples having another
MDT for the same gene.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate switches in condition 1 compared to condition 2</span></span>
<span><span class="va">Cond1_Cond2_Switches</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/switch_calculator.html">switch_calculator</a></span><span class="op">(</span>cancer_ori <span class="op">=</span> <span class="va">cond1_mdts</span>, gtex_ori <span class="op">=</span> <span class="va">cond2_mdts</span>, exp_values_pcawg <span class="op">=</span> <span class="va">cond1_ENST_ENSG_nonredundant</span>, exp_values_gtex <span class="op">=</span> <span class="va">cond2_ENST_ENSG_nonredundant</span>, ensp_sequences <span class="op">=</span> <span class="va">ensg_enst_seq</span>, ensp_only <span class="op">=</span> <span class="va">ensp_only</span>, cutoff <span class="op">=</span> <span class="fl">0</span>, cutoff_rate <span class="op">=</span> <span class="fl">2</span>, cutoff_other_MDTs <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visalize the results. Note that you should have knitr R package installed.</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">Cond1_Cond2_Switches</span>,<span class="fl">5</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">'MDT Switching Events'</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<caption>MDT Switching Events</caption>
<colgroup>
<col width="14%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="4%">
<col width="9%">
<col width="8%">
<col width="15%">
<col width="4%">
</colgroup>
<thead><tr class="header">
<th align="left">SampleID</th>
<th align="left">ENSG</th>
<th align="left">dMDT</th>
<th align="left">ENST2_cancer</th>
<th align="right">TPM1_cancer</th>
<th align="right">TPM2_cancer</th>
<th align="right">enrichment</th>
<th align="right">p_value</th>
<th align="right">relative_cancer_exp</th>
<th align="right">relative_gtex_exp</th>
<th align="left">MDT_GTEx</th>
<th align="right">adj_p</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">tpm_kallisto_macular_retina-1</td>
<td align="left">ENSG00000020256</td>
<td align="left">ENST00000346617</td>
<td align="left">ENST00000216923</td>
<td align="right">3.342410</td>
<td align="right">1.644050</td>
<td align="right">2.033034</td>
<td align="right">0.0002441</td>
<td align="right">0.3602014</td>
<td align="right">0.1078287</td>
<td align="left">ENST00000216923</td>
<td align="right">0.0002604</td>
</tr>
<tr class="even">
<td align="left">tpm_kallisto_macular_retina-1</td>
<td align="left">ENSG00000062282</td>
<td align="left">ENST00000605099</td>
<td align="left">ENST00000228027</td>
<td align="right">0.936847</td>
<td align="right">0.345076</td>
<td align="right">2.714901</td>
<td align="right">0.0002441</td>
<td align="right">0.6704946</td>
<td align="right">0.0000000</td>
<td align="left">ENST00000376262,ENST00000228027</td>
<td align="right">0.0002604</td>
</tr>
<tr class="odd">
<td align="left">tpm_kallisto_macular_retina-1</td>
<td align="left">ENSG00000070669</td>
<td align="left">ENST00000422745</td>
<td align="left">ENST00000455086</td>
<td align="right">13.171600</td>
<td align="right">6.408810</td>
<td align="right">2.055233</td>
<td align="right">0.0002441</td>
<td align="right">0.2935822</td>
<td align="right">0.0937156</td>
<td align="left">ENST00000455086</td>
<td align="right">0.0002604</td>
</tr>
<tr class="even">
<td align="left">tpm_kallisto_macular_retina-1</td>
<td align="left">ENSG00000083454</td>
<td align="left">ENST00000345901</td>
<td align="left">ENST00000552456</td>
<td align="right">5.319620</td>
<td align="right">1.283410</td>
<td align="right">4.144911</td>
<td align="right">0.0002441</td>
<td align="right">0.6319887</td>
<td align="right">0.0000000</td>
<td align="left">ENST00000552456</td>
<td align="right">0.0002604</td>
</tr>
<tr class="odd">
<td align="left">tpm_kallisto_macular_retina-1</td>
<td align="left">ENSG00000089597</td>
<td align="left">ENST00000356638</td>
<td align="left">ENST00000346178</td>
<td align="right">48.838600</td>
<td align="right">19.508100</td>
<td align="right">2.503504</td>
<td align="right">0.0002441</td>
<td align="right">0.5505801</td>
<td align="right">0.2352723</td>
<td align="left">ENST00000346178</td>
<td align="right">0.0002604</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="calculate-functional-impact-of-transcript-switches-by-integrating-the-transcript-interaction-network">Calculate Functional Impact of Transcript Switches by Integrating
the Transcript Interaction Network<a class="anchor" aria-label="anchor" href="#calculate-functional-impact-of-transcript-switches-by-integrating-the-transcript-interaction-network"></a>
</h3>
<div class="section level4">
<h4 id="isoform-interaction-network">Isoform Interaction Network<a class="anchor" aria-label="anchor" href="#isoform-interaction-network"></a>
</h4>
<p>Different transcripts of a gene can have varying exons, leading to
changes in protein domains. These changes may disrupt interactions with
protein interaction partners. To understand the functional impact of
these switching events, we incorporate the transcript interaction
network into the switching results. <strong>MDTToolset</strong> provides
Isoform Network, IsoNetwork_v12, which is based on STRING v12. This
network has the following columns. <em>ENSG</em>: Ensembl gene ID
<em>ENSP</em>: Ensembl protein ID <em>ENST</em>: Ensembl transcript ID
<em>STRINGensp</em>: STRING protein ID, considered canonical
<em>STRINGintN</em>: Number of proteins STRINGensp interacts with in the
STRING database <em>ExistIntN</em>: Number of interactions ENST is
involved in <em>MissIntN</em>: Number of interactions ENST has lost
<em>RelMissIntN</em>: Percentage of interaction loss <em>ExistInts</em>:
Existing interactions of ENST <em>MissInts</em>: Lost interactions of
ENST</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">IsoNetwork_v12</span> <span class="op">&lt;-</span> <span class="fu">MDTToolset</span><span class="fu">::</span><span class="va">IsoNetwork_v12_subsetData</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">IsoNetwork_v12</span>,<span class="fl">5</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">'Isoform Interaction Network'</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Isoform Interaction Network</caption>
<colgroup>
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="31%">
<col width="61%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">ENSG</th>
<th align="left">ENSP</th>
<th align="left">ENST</th>
<th align="left">STRINGensp</th>
<th align="right">STRINGintN</th>
<th align="right">ExistIntN</th>
<th align="right">MissIntN</th>
<th align="left">RelMissIntN</th>
<th align="left">ExistInts</th>
<th align="left">MissInts</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">2507</td>
<td align="left">ENSG00000017260</td>
<td align="left">ENSP00000432956</td>
<td align="left">ENST00000533801</td>
<td align="left">ENSP00000421326</td>
<td align="right">39</td>
<td align="right">6</td>
<td align="right">1</td>
<td align="left">0.143</td>
<td align="left">PF00122:E1-E2_ATPase:167-362:PF03381:CDC50:ENSP00000230461:402:1,PF00122:E1-E2_ATPase:167-362:PF00122:E1-E2_ATPase:ENSP00000263519:599:1,PF00122:E1-E2_ATPase:167-362:PF00122:E1-E2_ATPase:ENSP00000349877:425:1,PF00122:E1-E2_ATPase:167-362:PF00122:E1-E2_ATPase:ENSP00000350310:600:1,PF00122:E1-E2_ATPase:167-362:PF00122:E1-E2_ATPase:ENSP00000353414:629:1,PF00122:E1-E2_ATPase:167-362:PF00122:E1-E2_ATPase:ENSP00000392043:631:1</td>
<td align="left">PF00689:Cation_ATPase_C:760-932:PF05366:Sarcolipin:ENSP00000435380:413:1</td>
</tr>
<tr class="even">
<td align="left">2573</td>
<td align="left">ENSG00000018625</td>
<td align="left">ENSP00000376066</td>
<td align="left">ENST00000392233</td>
<td align="left">ENSP00000354490</td>
<td align="right">102</td>
<td align="right">7</td>
<td align="right">13</td>
<td align="left">0.650</td>
<td align="left">PF00122:E1-E2_ATPase:161-352:PF00122:E1-E2_ATPase:ENSP00000218548:725:1,PF00122:E1-E2_ATPase:161-352:PF00122:E1-E2_ATPase:ENSP00000332756:414:1,PF00122:E1-E2_ATPase:161-352:PF00122:E1-E2_ATPase:ENSP00000353414:421:1,PF00122:E1-E2_ATPase:161-352:PF00122:E1-E2_ATPase:ENSP00000357060:936:1,PF00122:E1-E2_ATPase:161-352:PF00122:E1-E2_ATPase:ENSP00000440045:873:1,PF00122:E1-E2_ATPase:161-352:PF00122:E1-E2_ATPase:ENSP00000444688:978:1,PF00122:E1-E2_ATPase:161-352:PF00122:E1-E2_ATPase:ENSP00000445306:936:1</td>
<td align="left">PF00689:Cation_ATPase_C:796-1005:PF00287:Na_K-ATPase:ENSP00000218008:960:1,PF00689:Cation_ATPase_C:796-1005:PF00287:Na_K-ATPase:ENSP00000250111:993:1,PF00689:Cation_ATPase_C:796-1005:PF02038:ATP1G1_PLM_MAT8:ENSP00000270310:599:1,PF00689:Cation_ATPase_C:796-1005:PF00287:Na_K-ATPase:ENSP00000286371:978:1,PF00689:Cation_ATPase_C:796-1005:PF02038:ATP1G1_PLM_MAT8:ENSP00000292079:978:1,PF00689:Cation_ATPase_C:796-1005:PF00287:Na_K-ATPase:ENSP00000334216:770:1,PF00689:Cation_ATPase_C:796-1005:PF04272:Phospholamban:ENSP00000350132:860:1,PF00689:Cation_ATPase_C:796-1005:PF00287:Na_K-ATPase:ENSP00000356790:996:1,PF00689:Cation_ATPase_C:796-1005:PF02038:ATP1G1_PLM_MAT8:ENSP00000442756:709:1,PF00689:Cation_ATPase_C:796-1005:PF02038:ATP1G1_PLM_MAT8:ENSP00000473361:558:1,PF00689:Cation_ATPase_C:796-1005:PF02038:ATP1G1_PLM_MAT8:ENSP00000473929:583:1,PF00689:Cation_ATPase_C:796-1005:PF02038:ATP1G1_PLM_MAT8:ENSP00000481244:946:1,PF00689:Cation_ATPase_C:796-1005:PF02038:ATP1G1_PLM_MAT8:ENSP00000482442:453:1</td>
</tr>
<tr class="odd">
<td align="left">91915</td>
<td align="left">ENSG00000204427</td>
<td align="left">ENSP00000418331</td>
<td align="left">ENST00000477462</td>
<td align="left">ENSP00000379282</td>
<td align="right">56</td>
<td align="right">0</td>
<td align="right">12</td>
<td align="left">1.000</td>
<td align="left">-</td>
<td align="left">PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000247706:448:1,PF00561:Abhydrolase_1:280-410:PF12146:Hydrolase_4:ENSP00000250974:560:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000268129:571:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000273359:541:1,PF00561:Abhydrolase_1:280-410:PF12697:Abhydrolase_6:ENSP00000273596:461:1,PF00561:Abhydrolase_1:280-410:PF12697:Abhydrolase_6:ENSP00000289119:649:1,PF00561:Abhydrolase_1:280-410:PF12146:Hydrolase_4:ENSP00000302657:415:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000326491:501:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000336693:577:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000365063:589:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000365725:855:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000414558:545:1</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">ENSG00000000003</td>
<td align="left">ENSP00000362111</td>
<td align="left">ENST00000373020</td>
<td align="left">ENSP00000362111</td>
<td align="right">17</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">-</td>
<td align="left">-</td>
<td align="left">-</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">ENSG00000000003</td>
<td align="left">ENSP00000482130</td>
<td align="left">ENST00000612152</td>
<td align="left">ENSP00000362111</td>
<td align="right">17</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">-</td>
<td align="left">-</td>
<td align="left">-</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># read where the Pfam</span></span>
<span><span class="va">ENSG_ENST_ENSP_Pfam</span> <span class="op">&lt;-</span> <span class="fu">MDTToolset</span><span class="fu">::</span><span class="va">ENSG_ENST_ENSP_Pfam</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="calculate-the-functional-impact">Calculate the Functional Impact<a class="anchor" aria-label="anchor" href="#calculate-the-functional-impact"></a>
</h4>
<p>To understand if the interaction partners of ENSTs (lost or kept) are
expressed in the sample, we implemented a function
<code><a href="../reference/buildIsoNet.html">buildIsoNet()</a></code> which takes four inputs:
<em>dMDT_network_info</em>: Merged database: switching results and
Isoform Network <em>ensembl_data</em>: ENSG_ENST_ENSP_Pfam
<em>kallisto_counts</em>: cond1_ENST_ENSG_nonredundant
<em>iso_network</em>: IsoNetwork_v12</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># we first merge the switching results with Isoform Network</span></span>
<span><span class="va">dMDTs_IsoNetwork</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">Cond1_Cond2_Switches</span>, <span class="va">IsoNetwork_v12</span>, by.x<span class="op">=</span><span class="st">'dMDT'</span>, by.y<span class="op">=</span><span class="st">'ENST'</span><span class="op">)</span></span>
<span><span class="va">dMDTs_IsoNetwork_int</span> <span class="op">&lt;-</span> <span class="va">dMDTs_IsoNetwork</span><span class="op">[</span><span class="va">dMDTs_IsoNetwork</span><span class="op">$</span><span class="va">MissInts</span> <span class="op">!=</span> <span class="st">'-'</span> <span class="op">|</span> <span class="va">dMDTs_IsoNetwork</span><span class="op">$</span><span class="va">ExistInts</span> <span class="op">!=</span> <span class="st">'-'</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">Switching_Events_Network</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/buildIsoNet.html">buildIsoNet</a></span><span class="op">(</span>dMDT_network_info <span class="op">=</span> <span class="va">dMDTs_IsoNetwork_int</span>, ensembl_data <span class="op">=</span> <span class="va">ENSG_ENST_ENSP_Pfam</span>, kallisto_counts <span class="op">=</span> <span class="va">cond1_ENST_ENSG_nonredundant</span>, iso_network <span class="op">=</span> <span class="va">IsoNetwork_v12</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">Switching_Events_Network</span>,<span class="fl">2</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">'Interaction Lost of dMDTs'</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Interaction Lost of dMDTs</caption>
<colgroup>
<col width="0%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="1%">
<col width="1%">
<col width="2%">
<col width="0%">
<col width="1%">
<col width="1%">
<col width="1%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="0%">
<col width="56%">
<col width="10%">
<col width="1%">
<col width="10%">
<col width="1%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">dMDT</th>
<th align="left">SampleID</th>
<th align="left">ENSG.x</th>
<th align="left">ENST2_cancer</th>
<th align="right">TPM1_cancer</th>
<th align="right">TPM2_cancer</th>
<th align="right">enrichment</th>
<th align="right">p_value</th>
<th align="right">relative_cancer_exp</th>
<th align="right">relative_gtex_exp</th>
<th align="left">MDT_GTEx</th>
<th align="right">adj_p</th>
<th align="left">ENSG.y</th>
<th align="left">ENSP</th>
<th align="left">STRINGensp</th>
<th align="right">STRINGintN</th>
<th align="right">ExistIntN</th>
<th align="right">MissIntN</th>
<th align="left">RelMissIntN</th>
<th align="left">ExistInts</th>
<th align="left">MissInts</th>
<th align="left">ensp_expressed_missed</th>
<th align="left">ensp_expressed_kept</th>
<th align="left">ensp_expressed_missed_in_sample</th>
<th align="left">int_losts_in_mdt</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">3</td>
<td align="left">ENST00000477462</td>
<td align="left">tpm_kallisto_macular_retina-2</td>
<td align="left">ENSG00000204427</td>
<td align="left">ENST00000496579</td>
<td align="right">0.517667</td>
<td align="right">0.171691</td>
<td align="right">3.015109</td>
<td align="right">0.0002441</td>
<td align="right">0.609764</td>
<td align="right">0</td>
<td align="left">ENST00000395952,ENST00000482224</td>
<td align="right">0.0002604</td>
<td align="left">ENSG00000204427</td>
<td align="left">ENSP00000418331</td>
<td align="left">ENSP00000379282</td>
<td align="right">56</td>
<td align="right">0</td>
<td align="right">12</td>
<td align="left">1.000</td>
<td align="left">-</td>
<td align="left">PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000247706:448:1,PF00561:Abhydrolase_1:280-410:PF12146:Hydrolase_4:ENSP00000250974:560:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000268129:571:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000273359:541:1,PF00561:Abhydrolase_1:280-410:PF12697:Abhydrolase_6:ENSP00000273596:461:1,PF00561:Abhydrolase_1:280-410:PF12697:Abhydrolase_6:ENSP00000289119:649:1,PF00561:Abhydrolase_1:280-410:PF12146:Hydrolase_4:ENSP00000302657:415:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000326491:501:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000336693:577:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000365063:589:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000365725:855:1,PF00561:Abhydrolase_1:280-410:PF00561:Abhydrolase_1:ENSP00000414558:545:1</td>
<td align="left">Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed</td>
<td align="left">No kept int</td>
<td align="left">Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed,Not_expressed</td>
<td align="left">MDT_is_not_found</td>
</tr></tbody>
</table>
</div>
</div>
<div class="section level3">
<h3 id="visualize-network-of-interaction-lost">Visualize Network of Interaction Lost<a class="anchor" aria-label="anchor" href="#visualize-network-of-interaction-lost"></a>
</h3>
<p><code><a href="../reference/visualize_network.html">visualize_network()</a></code> function will take two inputs: ENSP
id, and Isoform Network, IsoNetwork_v12. And it gives a graph showing
the network of the protein (of transript of interest) using igraph R
package.</p>
<div class="section level4">
<h4 id="example-1-all-interactions-are-lost">Example 1: All Interactions are Lost<a class="anchor" aria-label="anchor" href="#example-1-all-interactions-are-lost"></a>
</h4>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize_network.html">visualize_network</a></span><span class="op">(</span><span class="st">'ENSP00000418331'</span>, <span class="va">IsoNetwork_v12</span><span class="op">)</span></span></code></pre></div>
<p><img src="MDTToolset_switching-vignette_files/figure-html/drawNetwork-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; NULL</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="example-2-some-interactions-are-lost-while-some-are-kept">Example 2: Some interactions are lost while some are kept<a class="anchor" aria-label="anchor" href="#example-2-some-interactions-are-lost-while-some-are-kept"></a>
</h4>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize_network.html">visualize_network</a></span><span class="op">(</span><span class="st">'ENSP00000432956'</span>, <span class="va">IsoNetwork_v12</span><span class="op">)</span></span></code></pre></div>
<p><img src="MDTToolset_switching-vignette_files/figure-html/drawNetwork2-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; NULL</span></span>
<span><span class="fu"><a href="../reference/visualize_network.html">visualize_network</a></span><span class="op">(</span><span class="st">'ENSP00000376066'</span>, <span class="va">IsoNetwork_v12</span><span class="op">)</span></span></code></pre>
<p><img src="MDTToolset_switching-vignette_files/figure-html/drawNetwork2-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; NULL</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="check-expression-level-of-dmdtmdt-in-the-sample">Check Expression Level of dMDT:MDT in the sample<a class="anchor" aria-label="anchor" href="#check-expression-level-of-dmdtmdt-in-the-sample"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotExpression.html">plotExpression</a></span><span class="op">(</span><span class="st">'ENST00000346617'</span>, <span class="st">'ENST00000216923'</span>, <span class="st">'tpm_kallisto_macular_retina-1'</span>, <span class="va">cond1_ENST_ENSG_nonredundant</span>, <span class="va">cond2_ENST_ENSG_nonredundant</span><span class="op">)</span></span></code></pre></div>
<p><img src="MDTToolset_switching-vignette_files/figure-html/plotExpression-1.png" width="700"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tulay Karakulak.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
