<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MDTToolset">
<title>Introduction to Most Dominant Transcripts • MDTToolset</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to Most Dominant Transcripts">
<meta property="og:description" content="MDTToolset">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MDTToolset</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/MDTToolset_switching-vignette.html">MDT Switching Event and Its Functional Impact</a>
    <a class="dropdown-item" href="../articles/MDTToolset-vignette.html">Introduction to Most Dominant Transcripts</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/KarakulakTulay/MDTToolset/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Introduction to Most Dominant Transcripts</h1>
            <h3 data-toc-skip class="subtitle"><em>Detection of Most
Dominant Transcripts</em></h3>
                        <h4 data-toc-skip class="author">Tülay
Karakulak</h4>
            
            <h4 data-toc-skip class="date">2024-05-09</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/KarakulakTulay/MDTToolset/blob/HEAD/vignettes/MDTToolset-vignette.Rmd" class="external-link"><code>vignettes/MDTToolset-vignette.Rmd</code></a></small>
      <div class="d-none name"><code>MDTToolset-vignette.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>This guide provides a step-by-step workflow for identifying the Most
Dominant Transcripts (MDTs) within a bulk RNA-sequencing dataset. The
<strong>MDTToolset</strong> package is designed to work with outputs
generated by various RNA-Seq quantification tools, including, but not
limited to, Salmon and Kallisto. For the accurate detection of MDTs,
users must utilize normalized data values (e.g., TPM values) for bulk
RNA-seq data.</p>
</div>
<div class="section level2">
<h2 id="load-mdttoolset">Load MDTToolset<a class="anchor" aria-label="anchor" href="#load-mdttoolset"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KarakulakTulay/MDTToolset" class="external-link">MDTToolset</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="prepare-data">Prepare Data<a class="anchor" aria-label="anchor" href="#prepare-data"></a>
</h3>
<p>In this section, we utilize the TPM outputs from the Kallisto tool,
originating from an (RNA-Seq experiment) [<a href="https://www.nature.com/articles/s41597-020-0541-4" class="external-link uri">https://www.nature.com/articles/s41597-020-0541-4</a>]. The
data is openly accessible and available for download. The MDTToolset
includes two conditions as example cases for users. If you already have
the quantification data prepared for each condition, you may proceed to
the next step. Ensure that the first two column names are ‘ENST’ and
‘ENSG’, respectively, followed by the sample names as shown below.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## read TPM data </span></span>
<span><span class="va">cond1</span> <span class="op">&lt;-</span> <span class="fu">MDTToolset</span><span class="fu">::</span><span class="va">cond1</span> <span class="co"># molecular retina</span></span>
<span><span class="va">cond2</span> <span class="op">&lt;-</span> <span class="fu">MDTToolset</span><span class="fu">::</span><span class="va">cond2</span> <span class="co"># peripheral retina </span></span>
<span></span>
<span><span class="co"># visalize the dataframes. Note that you should have knitr R package installed.</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cond1</span>,<span class="fl">2</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">'TPM Matrix'</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>TPM Matrix</caption>
<colgroup>
<col width="4%">
<col width="7%">
<col width="8%">
<col width="8%">
<col width="8%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
</colgroup>
<thead><tr class="header">
<th align="left">ENST</th>
<th align="right">tpm_kallisto_macular_retina.1</th>
<th align="right">tpm_kallisto_macular_retina.10</th>
<th align="right">tpm_kallisto_macular_retina.11</th>
<th align="right">tpm_kallisto_macular_retina.12</th>
<th align="right">tpm_kallisto_macular_retina.2</th>
<th align="right">tpm_kallisto_macular_retina.3</th>
<th align="right">tpm_kallisto_macular_retina.4</th>
<th align="right">tpm_kallisto_macular_retina.5</th>
<th align="right">tpm_kallisto_macular_retina.6</th>
<th align="right">tpm_kallisto_macular_retina.7</th>
<th align="right">tpm_kallisto_macular_retina.8</th>
<th align="right">tpm_kallisto_macular_retina.9</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENST00000000233</td>
<td align="right">189.3070</td>
<td align="right">202.8500</td>
<td align="right">203.376</td>
<td align="right">232.7330</td>
<td align="right">191.1680</td>
<td align="right">184.5650</td>
<td align="right">195.5500</td>
<td align="right">228.8620</td>
<td align="right">202.2350</td>
<td align="right">248.3690</td>
<td align="right">196.5360</td>
<td align="right">235.9530</td>
</tr>
<tr class="even">
<td align="left">ENST00000000412</td>
<td align="right">43.7888</td>
<td align="right">52.5654</td>
<td align="right">52.852</td>
<td align="right">65.5688</td>
<td align="right">59.9673</td>
<td align="right">53.0147</td>
<td align="right">64.1605</td>
<td align="right">48.3739</td>
<td align="right">60.3188</td>
<td align="right">62.3249</td>
<td align="right">46.5275</td>
<td align="right">63.5542</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cond2</span>,<span class="fl">2</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">'TPM Matrix'</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>TPM Matrix</caption>
<colgroup>
<col width="3%">
<col width="7%">
<col width="8%">
<col width="8%">
<col width="8%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
</colgroup>
<thead><tr class="header">
<th align="left">ENST</th>
<th align="right">tpm_kallisto_peripheral_retina.1</th>
<th align="right">tpm_kallisto_peripheral_retina.10</th>
<th align="right">tpm_kallisto_peripheral_retina.11</th>
<th align="right">tpm_kallisto_peripheral_retina.12</th>
<th align="right">tpm_kallisto_peripheral_retina.2</th>
<th align="right">tpm_kallisto_peripheral_retina.3</th>
<th align="right">tpm_kallisto_peripheral_retina.4</th>
<th align="right">tpm_kallisto_peripheral_retina.5</th>
<th align="right">tpm_kallisto_peripheral_retina.6</th>
<th align="right">tpm_kallisto_peripheral_retina.7</th>
<th align="right">tpm_kallisto_peripheral_retina.8</th>
<th align="right">tpm_kallisto_peripheral_retina.9</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENST00000000233</td>
<td align="right">121.7770</td>
<td align="right">136.121</td>
<td align="right">154.4650</td>
<td align="right">169.1770</td>
<td align="right">148.1950</td>
<td align="right">174.5520</td>
<td align="right">147.8110</td>
<td align="right">147.3530</td>
<td align="right">132.3580</td>
<td align="right">138.844</td>
<td align="right">135.4470</td>
<td align="right">165.8010</td>
</tr>
<tr class="even">
<td align="left">ENST00000000412</td>
<td align="right">41.0036</td>
<td align="right">54.297</td>
<td align="right">55.0917</td>
<td align="right">62.8938</td>
<td align="right">59.8682</td>
<td align="right">49.5176</td>
<td align="right">61.5132</td>
<td align="right">48.0374</td>
<td align="right">63.8108</td>
<td align="right">60.491</td>
<td align="right">52.6708</td>
<td align="right">56.7923</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="prepare-files-to-remove-the-redundant-transcripts">Prepare Files to Remove the Redundant Transcripts<a class="anchor" aria-label="anchor" href="#prepare-files-to-remove-the-redundant-transcripts"></a>
</h3>
<p>This step is optional but highly recommended.</p>
<p>A challenge in transcript quantification is the presence of redundant
transcripts, which are transcripts with identical sequences that yield
the same TPM values. This redundancy can obscure the identification of
Most Dominant Transcripts (MDTs) by preventing the detection of a
singular, dominant transcript for a given gene. To address this issue,
our initial step involves eliminating redundant transcripts from the
dataset. This process includes assigning TPM values to a single
representative transcript per redundant group, based on the comparison
of their protein sequences.</p>
<p>Priority is given to transcripts classified as MANE (Matched
Annotation from NCBI and EMBL-EBI) in the Ensembl database, designating
them as the primary transcript. In cases where no MANE transcript is
identified, a transcript is randomly selected to represent the group.
This step is particularly crucial when working with tools like Kallisto,
which may assign identical TPM values to redundant transcripts.
Following this, the calculation of MDTs proceeds using the reassigned
TPM values.</p>
<p>To be able to detect redundat transcripts, we need sequences of
transcripts. Fasta sequence of transcripts (cdna sequences) can be
downloaded from the designated databases (e.g Ensembl)[<a href="https://www.ensembl.org/info/data/ftp/index.html" class="external-link uri">https://www.ensembl.org/info/data/ftp/index.html</a>]. We
provide Ensembl version 104 as an example. In addition, we provide the
Ensembl MANE Select Data (v104) to help remove redundant transcripts
found in Kallisto output files. To download your version of the MANE
select, navigate to the Ensembl website for the version you used,
proceed to the BioMart website, select the fields <em>Gene Stable
ID</em>, <em>Transcript Stable ID</em>, and <em>RefSeq Match Transcript
(MANE Select)</em>, and download the TSV file. Please note that
<strong>The user should employ the same version that was used to
quantify their transcript abundances to ensure consistency.</strong> the
function <code><a href="../reference/prepare_Seq.html">prepare_Seq()</a></code> takes fasta and MANE file as input
and gives a dataframe showing ENSG (Gene), ENST (Transcript), and ENSP
(Protein), and MANE ID, and ENST Sequence.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># read MANE Select</span></span>
<span><span class="va">ensg_enst_ensp_mane_v104</span> <span class="op">&lt;-</span> <span class="fu">MDTToolset</span><span class="fu">::</span><span class="va">ensg_enst_ensp_mane_v104</span></span>
<span><span class="va">ensg_enst_ensp_mane_v104</span> <span class="op">&lt;-</span> <span class="va">ensg_enst_ensp_mane_v104</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># While downloading data from Biomart, the values might not be unique - make sure the rows are unique</span></span>
<span><span class="co"># prepare sequences and Mane select transcripts dataframe</span></span>
<span><span class="va">ensg_enst_seq</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepare_Seq.html">prepare_Seq</a></span><span class="op">(</span><span class="fu">MDTToolset</span><span class="fu">::</span><span class="va">Homo_sapiens.GRCh38.cdna.all</span>, <span class="va">ensg_enst_ensp_mane_v104</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Select transcripts having MANE ids</span></span>
<span><span class="va">enst_mane_iso</span> <span class="op">&lt;-</span> <span class="va">ensg_enst_ensp_mane_v104</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">ensg_enst_ensp_mane_v104</span><span class="op">$</span><span class="va">RefSeq.match.transcript..MANE.Select.</span><span class="op">)</span> <span class="op">|</span> <span class="va">ensg_enst_ensp_mane_v104</span><span class="op">$</span><span class="va">RefSeq.match.transcript..MANE.Select.</span> <span class="op">==</span> <span class="st">''</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">4</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<p>We merge the transcript TPM expression matrix file with the
ensg_enst_seq file that we have prepared in the previous step so that
the second column includes the ENSG ids.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cond1_ENST_ENSG</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">ensg_enst_seq</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>, <span class="va">cond1</span>, by<span class="op">=</span><span class="st">'ENST'</span><span class="op">)</span></span>
<span><span class="va">cond2_ENST_ENSG</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">ensg_enst_seq</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>, <span class="va">cond2</span>, by<span class="op">=</span><span class="st">'ENST'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visalize the dataframes. Note that you should have knitr R package installed.</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cond1_ENST_ENSG</span>,<span class="fl">2</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">'TPM Matrix'</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<caption>TPM Matrix</caption>
<colgroup>
<col width="4%">
<col width="4%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
</colgroup>
<thead><tr class="header">
<th align="left">ENST</th>
<th align="left">ENSG</th>
<th align="right">tpm_kallisto_macular_retina.1</th>
<th align="right">tpm_kallisto_macular_retina.10</th>
<th align="right">tpm_kallisto_macular_retina.11</th>
<th align="right">tpm_kallisto_macular_retina.12</th>
<th align="right">tpm_kallisto_macular_retina.2</th>
<th align="right">tpm_kallisto_macular_retina.3</th>
<th align="right">tpm_kallisto_macular_retina.4</th>
<th align="right">tpm_kallisto_macular_retina.5</th>
<th align="right">tpm_kallisto_macular_retina.6</th>
<th align="right">tpm_kallisto_macular_retina.7</th>
<th align="right">tpm_kallisto_macular_retina.8</th>
<th align="right">tpm_kallisto_macular_retina.9</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENST00000000233</td>
<td align="left">ENSG00000004059</td>
<td align="right">189.3070</td>
<td align="right">202.8500</td>
<td align="right">203.376</td>
<td align="right">232.7330</td>
<td align="right">191.1680</td>
<td align="right">184.5650</td>
<td align="right">195.5500</td>
<td align="right">228.8620</td>
<td align="right">202.2350</td>
<td align="right">248.3690</td>
<td align="right">196.5360</td>
<td align="right">235.9530</td>
</tr>
<tr class="even">
<td align="left">ENST00000000412</td>
<td align="left">ENSG00000003056</td>
<td align="right">43.7888</td>
<td align="right">52.5654</td>
<td align="right">52.852</td>
<td align="right">65.5688</td>
<td align="right">59.9673</td>
<td align="right">53.0147</td>
<td align="right">64.1605</td>
<td align="right">48.3739</td>
<td align="right">60.3188</td>
<td align="right">62.3249</td>
<td align="right">46.5275</td>
<td align="right">63.5542</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cond2_ENST_ENSG</span>,<span class="fl">2</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">'TPM Matrix'</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<caption>TPM Matrix</caption>
<colgroup>
<col width="3%">
<col width="3%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
<col width="7%">
</colgroup>
<thead><tr class="header">
<th align="left">ENST</th>
<th align="left">ENSG</th>
<th align="right">tpm_kallisto_peripheral_retina.1</th>
<th align="right">tpm_kallisto_peripheral_retina.10</th>
<th align="right">tpm_kallisto_peripheral_retina.11</th>
<th align="right">tpm_kallisto_peripheral_retina.12</th>
<th align="right">tpm_kallisto_peripheral_retina.2</th>
<th align="right">tpm_kallisto_peripheral_retina.3</th>
<th align="right">tpm_kallisto_peripheral_retina.4</th>
<th align="right">tpm_kallisto_peripheral_retina.5</th>
<th align="right">tpm_kallisto_peripheral_retina.6</th>
<th align="right">tpm_kallisto_peripheral_retina.7</th>
<th align="right">tpm_kallisto_peripheral_retina.8</th>
<th align="right">tpm_kallisto_peripheral_retina.9</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENST00000000233</td>
<td align="left">ENSG00000004059</td>
<td align="right">121.7770</td>
<td align="right">136.121</td>
<td align="right">154.4650</td>
<td align="right">169.1770</td>
<td align="right">148.1950</td>
<td align="right">174.5520</td>
<td align="right">147.8110</td>
<td align="right">147.3530</td>
<td align="right">132.3580</td>
<td align="right">138.844</td>
<td align="right">135.4470</td>
<td align="right">165.8010</td>
</tr>
<tr class="even">
<td align="left">ENST00000000412</td>
<td align="left">ENSG00000003056</td>
<td align="right">41.0036</td>
<td align="right">54.297</td>
<td align="right">55.0917</td>
<td align="right">62.8938</td>
<td align="right">59.8682</td>
<td align="right">49.5176</td>
<td align="right">61.5132</td>
<td align="right">48.0374</td>
<td align="right">63.8108</td>
<td align="right">60.491</td>
<td align="right">52.6708</td>
<td align="right">56.7923</td>
</tr>
</tbody>
</table>
<p>As transcripts with the same sequences might have different peptide
lengths, we aim to select the ones with the longest sequence when there
is a redundancy. This information cannot be accessed directly via
Ensembl Biomart, this we need to download the transcripts’ peptide
sequences and then calculate the length of the peptides. We provide
Ensembl version 104 as an example. In case you want to prepare a
different version, download ENSG, ENST, peptide sequences from Biomart
and use the following function from MDTToolset:</p>
<p>df_biomart_seq_length &lt;-
MDTToolset::MDTToolset::prepare_seq_length(‘mart_export.txt’) The file
‘mart_export.txt’ is in the fasta file format:
&gt;ENSG0000000XXXX|ENST0000000KKK
MQRSPLEKASVVSKLFFSWTRPILRKGYRQRLELSDIYQIPSVDSADNLSEKLEREWDRE
LASKKNPKLINALRRCFFWRFMFYGIFLYLGEVTKAVQPLLLGRIIASYDPDNKEERSIA IYLG*
&gt;ENSG0000000YYYY|ENST0000000ZZZ
MTAEEMKATESGAQSAPLPMEGVDISPKQDEGVLKVIKREGTGTEMPMIGDRVFVHYTGW
LLDGTKFDSSLD</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example dataframe for df_biomart_seq_length </span></span>
<span><span class="va">df_biomart_seq_length</span> <span class="op">&lt;-</span> <span class="fu">MDTToolset</span><span class="fu">::</span><span class="va">df_biomart_seq_selection</span></span>
<span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">df_biomart_seq_length</span>,<span class="fl">2</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">"Transcripts' Peptide Seq Length"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Transcripts’ Peptide Seq Length</caption>
<colgroup>
<col width="48%">
<col width="21%">
<col width="21%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">ENSG</th>
<th align="left">ENST</th>
<th align="right">Length</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000001626|ENST00000003084</td>
<td align="left">ENSG00000001626</td>
<td align="left">ENST00000003084</td>
<td align="right">1481</td>
</tr>
<tr class="even">
<td align="left">ENSG00000004478|ENST00000001008</td>
<td align="left">ENSG00000004478</td>
<td align="left">ENST00000001008</td>
<td align="right">460</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="remove-redundant-transcripts">Remove Redundant Transcripts<a class="anchor" aria-label="anchor" href="#remove-redundant-transcripts"></a>
</h3>
<p><code>remove_redundant()</code> function takes three inputs,
transcript TPM file (having ENST and ENSG columns), ENST sequence file,
and Mane select file prepared in the previous step. If there is more
than one transcript of a gene with the same sequences, one of them is
removed, its TPM value is added to the selected transcript.
Additionally, if there is a transcript assigned as <code>MANE</code>,
this transcript is prioritized over the other ones. If there is no MANE
selects for the redundant transcripts, the transcripts with the longest
peptide sequence is chosen. The output will be the same as previous TPM
file without redundant ones.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Example Usage</span></span>
<span><span class="co">#cond1_ENST_ENSG_nonredundant &lt;- remove_redundants(data_matrix = cond1_ENST_ENSG, enst_ensp_sequences = ensg_enst_seq, mane_select =  NULL, peptide_lengths = NULL)</span></span>
<span><span class="va">cond1_ENST_ENSG_nonredundant</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_redundants.html">remove_redundants</a></span><span class="op">(</span><span class="va">cond1_ENST_ENSG</span>, <span class="va">ensg_enst_seq</span>, <span class="va">enst_mane_iso</span>, <span class="va">df_biomart_seq_length</span><span class="op">)</span></span>
<span><span class="va">cond2_ENST_ENSG_nonredundant</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_redundants.html">remove_redundants</a></span><span class="op">(</span><span class="va">cond2_ENST_ENSG</span>, <span class="va">ensg_enst_seq</span>, <span class="va">enst_mane_iso</span>, <span class="va">df_biomart_seq_length</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="find-mdts">Find MDTs<a class="anchor" aria-label="anchor" href="#find-mdts"></a>
</h3>
<p>The next step involves identifying the Most Dominant Transcripts
(MDTs) in the dataset. The find_mdts() function calculates MDTs based on
two criteria: a user-defined cutoff and min_exp. The cutoff parameter
specifies the enrichment rate between the expressions of the first and
second-ranked transcripts. min_exp determines the minimum expression
level that transcripts must meet to be considered as MDTs.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cond1_mdts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find_mdts.html">find_mdts</a></span><span class="op">(</span><span class="va">cond1_ENST_ENSG_nonredundant</span>, cutoff <span class="op">=</span> <span class="fl">2</span>, min_exp <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="va">cond2_mdts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find_mdts.html">find_mdts</a></span><span class="op">(</span><span class="va">cond2_ENST_ENSG_nonredundant</span>, cutoff <span class="op">=</span> <span class="fl">2</span>, min_exp <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="visualize-count-of-mdts">Visualize count of MDTs<a class="anchor" aria-label="anchor" href="#visualize-count-of-mdts"></a>
</h3>
<p>We provide a basic boxplot plot using ggplot to plot the number of
MDTs in each condition and also bar graph to visualize number of MDTs in
each sample. ### visualize the number of MDTs in each condition</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Distribution of number of MDTs in each sample</span></span>
<span><span class="fu"><a href="../reference/MDT_boxplot.html">MDT_boxplot</a></span><span class="op">(</span><span class="va">cond1_mdts</span>, <span class="va">cond2_mdts</span>, title <span class="op">=</span> <span class="st">"Distribution of Number of MDTs"</span>, x_axis_label <span class="op">=</span> <span class="st">"Condition"</span>, y_axis_label <span class="op">=</span> <span class="st">"Number of MDT"</span><span class="op">)</span></span></code></pre></div>
<p><img src="MDTToolset-vignette_files/figure-html/plotMDTcounts-1.png" width="700"></p>
<div class="section level4">
<h4 id="visualize-the-number-of-mdts-in-each-sample">visualize the number of MDTs in each sample<a class="anchor" aria-label="anchor" href="#visualize-the-number-of-mdts-in-each-sample"></a>
</h4>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Distribution of number of MDTs in each sample</span></span></code></pre></div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Tulay Karakulak.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
